sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AzureKeyCredentialPolicy.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * Pipeline policy that uses an {@link AzureKeyCredential} to set the specified header which authorizes requests.
+ *
+ * <p>
+ * Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
+ * an exception will be thrown to prevent leaking the key.
+ */
+public final class AzureKeyCredentialPolicy implements HttpPipelinePolicy {
+    private final ClientLogger logger = new ClientLogger(AzureKeyCredentialPolicy.class);
+
+    private final String header;
+    private final AzureKeyCredential credential;
+
+    /**
+     * Creates a policy that uses the passed {@link AzureKeyCredential} to set the specified header.
+     *
+     * @param header The header that will be set to the {@link AzureKeyCredential} key value.
+     * @param credential The {@link AzureKeyCredential} containing the authorization key to use.
+     * @throws NullPointerException If {@code header} or {@code credential} is {@code null}.
+     * @throws IllegalArgumentException If {@code header} is empty.
+     */
+    public AzureKeyCredentialPolicy(String header, AzureKeyCredential credential) {
```

Comment:
Alan, on Python land, Johan is suggesting to change the parameter name from `header` to `name` so whenever we have a service that uses query keys instead of header keys, this name will work.
I am considering changing it in .NET too so we are consistent
More info here: https://github.com/Azure/azure-sdk-for-python/pull/10509#discussion_r398744760

What do you think?

Reply:
Works with me, this is a non-breaking change for Java since we don't have named arguments.

Reply:
Though if we have plans to support both headers and query parameters for the passed key Java may want to change this interface a bit to have a flag such as `isHeaderKey` and context switch on that to either add it as a header or a query parameter. This could also be added later as a non-breaking change.